rust_src/src/fns.rs

```diff
@@ -571,4 +573,29 @@ pub fn nconc(args: &mut [LispObject]) -> LispObject {
     val
 }
 
+/// Return a copy of a list, vector, string, char-table or record.
+/// The elements of a list, vector or record are not copied; they are
+/// shared with the original.
+/// If the original sequence is empty, this function may return
+/// the same empty object instead of its copy.
+#[lisp_fn]
+pub fn copy_sequence(mut arg: LispObject) -> LispObject {
+    if arg.is_nil() {
+        arg
+    } else if arg.is_record() {
+        record(arg.force_vectorlike_slots().as_mut_slice())
+    } else if arg.is_char_table() {
+        unsafe { copy_char_table(arg) }
+    } else if let Some(boolvec) = arg.as_bool_vector() {
+        let nbits = boolvec.len() as i64;
+        let mut new = unsafe { make_uninit_bool_vector(nbits).force_bool_vector() };
+        new.as_mut_slice().copy_from_slice(boolvec.as_slice());
+        new.into()
+    } else if !arg.is_cons() && !arg.is_vector() && !arg.is_string() {
```

Comment:
```suggestion
    } else if !(arg.is_cons() || arg.is_vector() || arg.is_string()) {
```
Maybe some deMorgen's here?

Reply:
deMorgen's?

Reply:
I typo'ed. It is De Morgan's.
https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
essentially: `!A && !B == !(A || B)` and vice versa. Invert the operator, invert the checks, move the 'not' to the outside.

Reply:
Aha. Interesting read.